WebContent/WEB-INF/jsp/collection.jsp

```diff
@@ -43,10 +44,28 @@
 	    var collIndex = document.mainForm.category.selectedIndex;
 		var parentIndex = document.mainForm.parentCollection.selectedIndex;  
 	    if(parentIndex != 0){
-	    	if (category.options[collIndex].value == parentCollection.options[parentIndex].value) {
-	    		alert("Parent collection can't be the same collection. Please select another parent collection.");
-				return false;
-	    	}
+	    	var collTypeValue = document.mainForm.collType.options[collTypeIndex].value;
+	    	var parentColl = document.mainForm.parentCollection.options[parentIndex].text;
+	    	var parentCollType = parentColl.substring(parentColl.lastIndexOf("[")+1, parentColl.lastIndexOf("]"));
+
+    		var error = "";
+    		if (category.options[collIndex].value == parentCollection.options[parentIndex].value) {
+	    		error = "Parent collection can't be the same collection.";
+	   		} else if (collTypeValue == collTypes[2]) { // ProvenanceCollectionPart
+	   			if (parentCollType != collTypes[1]) 
+	   				error = "A " + collTypes[2] + " can only have a " + collTypes[1] + " parent.";
+	    	} else if (collTypeValue == collTypes[1]) { // ProvenanceCollection
+	   			if (parentCollType != collTypes[0])
+    				error = "A " + collTypes[1] + " can only have a " + collTypes[0] + " parent.";
```

Comment:
This new update looks great, thanks @lsitu 

Does this mean we're enforcing a ProvenanceCollection to have an AssembledCollection parent? If so, we probably need to change that so the parent association is optional for a ProvenanceCollection. Many ProvenanceCollections will never have a parent.